griptape/loaders/base_text_loader.py

```diff
@@ -43,12 +45,14 @@ def _text_to_artifacts(self, text: str) -> list[TextArtifact]:
 
         chunks = self.chunker.chunk(text) if self.chunker else [TextArtifact(text)]
 
-        if self.embedding_driver:
-            for chunk in chunks:
+        for chunk in chunks:
+            if self.embedding_driver:
                 chunk.generate_embedding(self.embedding_driver)
 
-        for chunk in chunks:
+            chunk.reference = self.reference
```

Comment:
Out of scope for this PR, but In the future, would we ever want to add additional info to chunk references, like offset into document or something? Doesn't seem that difficult to do from what's added in this PR, but just curious if it had crossed your mind at all

Reply:
Interesting! Yeah, I think that would be very cool, especially for some things that can be split up into logical sections more easily (e.g., markdown docs).